### ISSUE TITLE
Non-animated tab stack

### ISSUE THREAD

**Issue by kyle-ssg** [ajsmth/earhart#6]
So I think a common use case will be to have tabs that don't animate, I started creating a whatsapp clone to test this out a few weeks ago and noticed this. Also I think explaining how deep routing into nested navigators are intended to work would be quite handy as it's the only real complex concept.

For example, I may have a couple screens that can exist in tab a and tab b, if I want to deep link either on startup or clicking a push notification etc it seems that rather than explicitly setting tab this is clever and knows how to navigate sub Navigators?

**Comment by ajsmth** [comment]
for the first part - i usually use a switch for non animated tabs. 

for the next part, i’m not entirely clear on what you mean here, can you elaborate slightly on your example?

**Comment by kyle-ssg** [comment]
Ok, I'll see if there are any differences, I imagine with switch you wouldn't be able to persist where you were in the sub stack if you switched "tabs" without some bespoke state management.

As for the second I guess a couple examples, given this:

```
     <Router initialEntries={['/screen1']}>
            <Navigator>
                {({ params }: any) => {
                    return (
                        <>
                            <Switch>
                                <Route path="/tab1">
                                    <Stack>
                                        <Route path="/screen1"></Route>
                                        <Route path="/screen2"></Route>
                                    </Stack>
                                </Route>
                                <Route path="/tab2">
                                    <Stack>
                                        <Route path="/screen3"></Route>
                                        <Route path="/screen4"></Route>
                                    </Stack>
                                </Route>
                            </Switch>
                            <View style={{ flexDirection: 'row' }}>
                                <Link to="/tab1"></Link>
                                <Link to="/tab2"></Link>
                            </View>
                        </>
                    );
                }}
            </Navigator>
            <Location/>
        </Router>
```

1 - how would you initialise the app to tab 2 -> screen 4? And similarly for more nested complex apps with sub stacks.
2 - how would you link from screen 1 to screen 4?

**Comment by ajsmth** [comment]
> Ok, I'll see if there are any differences, I imagine with switch you wouldn't be able to persist where you were in the sub stack if you switched "tabs" without some bespoke state management.

This can be accounted for by using the "latest" flag on a link/navigate() call - for example a tab would pretty much always want to navigate to the last visited screen in that tab panel. Switch in this instance is using react-native-screens so it just removes the view from the hierarchy but persists it in memory

I suppose this is the "clever" part of the routing - when a navigator becomes inactive it doesn't update views at all so navigating back should restore it as it was, provided that the path is the same

Regarding the second part: 

1 - I would think navigating to `/tab2/screen4` should work
2 - A navigate(`/tab2/screen4`) should work, or a link in screen 1 with that same path

I notice that your paths aren't absolute, I'm wondering if thats for brevity's sake here but its a requirement for paths to be absolute to work together

**Comment by kyle-ssg** [comment]
RE the first part sounds good I'll probably just have to see it working.


The second part, unfortunately that didn't work if I understood what you were saying correctly. Given: <broken code>

By paths being absolute I assume that you mean a stack under a route must also contain the parents path i.e. /tab1/screenx in both Link paths and Route paths?

- Can't navigate to any screen within tab 2
- Clicking on a link for  /tab1/screen2 actually navigates me to /tab2/screen4

...

Edit: this is an example of both how to set the initial route as well as navigate between tabs.

```
    <Router initialEntries={["/tab2/screen5"]}>
            <Navigator>
                {({ params }: any) => {
                    return (
                        <>
                            <Switch>
                                <Route path="/tab1">
                                    <Navigator>
                                        <Stack>
                                            <Route path="/tab1/screen1">
                                                <Text>Screen 1</Text>
                                            </Route>
                                            <Route path="/tab1/screen2">
                                                <Text>Screen 2</Text>
                                            </Route>
                                        </Stack>
                                    </Navigator>
                                </Route>
                                <Route path="/tab2">
                                    <Navigator>
                                        <Stack>
                                            <Route path="/tab2/screen3">
                                                <Text>Screen 3</Text>
                                            </Route>
                                            <Route path="/tab2/screen4">
                                                <Text>Screen 4</Text>
                                            </Route>
                                            <Route path="/tab2/screen5">
                                                <Text>Screen 5</Text>
                                            </Route>
                                        </Stack>
                                    </Navigator>
                                </Route>
                            </Switch>
                            <View style={{ flexDirection: 'row' }}>
                                <Link to="/tab1/screen1">
                                    <Text>Tab 1</Text>
                                </Link>
                                <Link to="/tab2/screen5">
                                    <Text>Tab 2</Text>
                                </Link>
                            </View>
                        </>
                    );
                }}
            </Navigator>
            <Location/>
        </Router>
```

**Comment by ajsmth** [comment]
Yup, thats right about the paths. 

Looks like you're missing the navigator components as well, each screen container needs its own navigator. Maybe I should try to simplify the markup I think. What I mean: 
`<Navigator><Switch>...routes</Switch></Navigator>`

And same for stacks

**Comment by kyle-ssg** [comment]
Right ok I'm totally with you now! So a Navigator is basically just a container that manages the internal state and history of a screen container. Screen containers being switches, stacks and tabs.

To be honest the markup situation is totally fine so long as it all it is solid in managing deep hierarchies.

Thanks so much for responding to this, think I've got everything I need now to blast through some examples now. I've updated the above comment with a working example of setting the initial route and navigating between tabs.
